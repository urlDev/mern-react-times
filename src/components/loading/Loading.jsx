import DarkLoadingSrc from "../../assets/loadingLogo.svg";

import { LoadingContainer } from "./Loading.styles";

const Loading = ({ height, logoSrc, login }) => {
  return (
    <LoadingContainer height={height} login={login}>
      <div style={{ display: "flex", flexDirection: "column" }}>
        <img src={logoSrc || DarkLoadingSrc} alt="loading logo" />
      </div>
    </LoadingContainer>
  );
};

export default Loading;
