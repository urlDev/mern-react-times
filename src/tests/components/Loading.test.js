import { shallow } from "enzyme";

import Loading from "../../components/loading/Loading";

describe("Testing Loading component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Loading />);
  });

  test("Should render the component successfully", () => {
    expect(wrapper).toHaveLength(1);
  });

  test("Should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
