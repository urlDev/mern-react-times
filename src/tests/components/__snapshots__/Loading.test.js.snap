// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading component Should match the snapshot 1`] = `
<styled.div>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    <img
      alt="loading logo"
      src="loadingLogo.svg"
    />
  </div>
</styled.div>
`;
